File: examples/typescript/data/initial/friends.ts
import { factory } from '../../model/factory';
import { Friends } from '../../model/model';
import { DATAID } from '../dataid';

export function registerInitialFriends() {
  const f = factory;
  f.register(DATAID.TEST01_FRIENDS_1, () => {
    return {
      id: 0,
      name: 'Colon Salazar',
    } as Friends;
  });
  f.register(DATAID.TEST01_FRIENDS_2, () => {
    return {
      id: 1,
      name: 'French\nMcneil',
    } as Friends;
  });
  f.register(DATAID.TEST02_FRIENDS_2, () => {
    const data = f.inheritNode(DATAID.TEST01_FRIENDS_2) as Friends;
    data.name = 'French\nMcneil2';
    return data;
  });
  f.register(DATAID.TEST04_FRIENDS_3, () => {
    return {
      id: 2,
      name: 'Nestor Salinas',
    } as Friends;
  });
}

File: examples/golang/data/initial/friends.go
// +build test

package initial

import (
	"forcedata/examples/golang/model"
)

// データの識別子
const (
	Test01Friends1 model.DataID = "Test01Friends1"
	Test01Friends2 model.DataID = "Test01Friends2"
	Test02Friends2 model.DataID = "Test02Friends2"
	Test04Friends3 model.DataID = "Test04Friends3"
)

func RegisterFriends() {
	f := model.Factory
	f.Register(Test01Friends1, func() interface{} {
		return &model.Friends{
			Id:   0,
			Name: "Colon Salazar",
		}
	})
	f.Register(Test01Friends2, func() interface{} {
		return &model.Friends{
			Id:   1,
			Name: "French\nMcneil",
		}
	})
	f.Register(Test02Friends2, func() interface{} {
		data := f.InheritNode(Test01Friends2).(*model.Friends)
		data.Name = "French\nMcneil2"
		return data
	})
	f.Register(Test04Friends3, func() interface{} {
		return &model.Friends{
			Id:   2,
			Name: "Nestor Salinas",
		}
	})
}
